Sources/Sentry/SentryTracer.m

```diff
@@ -451,6 +451,10 @@ - (BOOL)hasUnfinishedChildSpansToWaitFor
 
 - (void)finishInternal
 {
+    // Keep existing status of auto generated transactions if set by the user.
+    if (!self.wasFinishCalled && self.status != kSentrySpanStatusUndefined) {
+        _finishStatus = self.status;
+    }
```

Comment:
h: The problem is that this solution does not check for an automatic transaction. 
Manual transaction also gets this behavior, and IMO it will be misleading if the user manually call `finishWithStatus` and get a different status because the `status` property was set before for some reason.

Reply:
I added `[self isAutoGeneratedTransaction] `. Thanks for pointing that out @brustolin.